### PR TITLE
#4 add an option to disable the opencv preview, drawing the edge and id …

### PR DESCRIPTION
**Pull request by WangQiHao-Charlie**
Issue #4 
1. Added an enableDraw flag to control whether detection outlines and text are drawn. This reduces unnecessary processing when visual output is not needed.
2. Modified the loop method to skip displaying and waiting operations if enableDraw is false, thus avoiding performance overhead during detection.
3. Included checks for empty frames and improved error messages for better debugging.

### CODE REVIEW

File: RobotCode/src/robotVision.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #include <iostream>
 #include <iomanip>
-
+#include <opencv2/opencv.hpp>
```
Comment (BanceDev) [inline RobotCode/src/robotVision.cpp:3]: redundant include, this header is in robotVision.h

File: RobotCode/src/robotVision.cpp [hunk 2]
```diff
@@ -39,12 +39,11 @@ RobotVision::RobotVision(int argc, char *argv[]) {
     // Initialize tag detector with options
     tf = tagStandard52h13_create();
 
-
     td = apriltag_detector_create();
     apriltag_detector_add_family(td, tf);
 
     if (errno == ENOMEM) {
-        printf("Unable to add family to detector due to insufficient memory to allocate the tag-family decoder with the default maximum hamming value of 2. Try choosing an alternative tag family.\n");
+        cerr << "Unable to add family to detector due to insufficient memory. Try choosing an alternative tag family.\n";
```
Comment (BanceDev) [inline RobotCode/src/robotVision.cpp:46]: we dont want to use GNU libc. Stick with C++ stdlib functions for logging

File: RobotCode/src/robotVision.cpp [hunk 3]
```diff
@@ -88,44 +129,18 @@ void RobotVision::loop() {
         zarray_t *detections = apriltag_detector_detect(td, &im);
 
         if (errno == EAGAIN) {
-            printf("Unable to create the %d threads requested.\n",td->nthreads);
+            cerr << "Unable to create the " << td->nthreads << " threads requested.\n";
```
Comment (BanceDev) [inline RobotCode/src/robotVision.cpp:132]: same thing as earlier keep C++ stdlib not GNU libc

**Review by BanceDev — CHANGES_REQUESTED** [review]
looks good on the whole, just tweak the stuff I commented and this should be good to go. If you have any questions please lmk

File: RobotCode/src/robotVision.cpp (hunk 2 above)
Comment (ncharlton02) [reply RobotCode/src/robotVision.cpp:46]: ? - cerr is a part of the C++ stdlib.

Comment (BanceDev) [reply RobotCode/src/robotVision.cpp:46]: image
ye its basically the iostream for the C stderr fd

**Comment by BanceDev** [comment]
I misread the original pr. I thought u changed cerr to printf not the other way around. 🫠 Ill just fix my mistake in a commit and then we can merge this.

**Pull request merged**